File: python_backend/fenParser/getBoard/get_pos_str.py
from custom_except import *
import re

def get_pos_str(fen):
    """fenParser task 0, return the position part of the fen string"""
    res = re.search(r'(((r|n|b|q|k|p|R|N|B|Q|K|P|[1-8])){1,8}/){7}((r|n|b|q|k|p|R|N|B|Q|K|P|[1-8]){1,8}){1}', fen)

    if res != None:
        pos_str = res.group(0)
        return pos_str
    else:
        print('error: position string not found')
        return -1


if __name__ == '__main__':

    # test 1:
    print('test 1:')
    pos_str = get_pos_str('rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1')
    print(pos_str)
    print('\n')

File: app (1)/src/game_logic/fenParser/getBoard/getPosStr.js

export function getPosStr(fen) {
    /*fenParser task 0, return the position part of the fen string**/
    let res = fen.match(`([a-zA-Z0-8]+/){7}([a-zA-Z0-8]+)`)
    if (res  !== null) {
        let posStr = res[0]
        return posStr
    }
    else {
        console.log('error: position string ! found')
        return -1
    }
}

// module.exports = getPosStr;

/** for node.js
if (require.main === module) {
    // test 1:
    console.log('test 1:')
    posStr = getPosStr('rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1')
    console.log(posStr)
    console.log('test 2:')
    posStr = getPosStr("r4rk1/1pp1qppp/2nbbn2/pP1pp3/3PP3/2NBBN2/P1P1QPPP/R3K2R w KQ a6 0 10")
    console.log(posStr)
    console.log('\n')

}
*/